frontends/web/src/components/password.jsx

```diff
@@ -129,8 +129,7 @@ export class PasswordSingleInput extends Component {
                     placeholder={placeholder}
                     onInput={this.handleFormChange}
                     onPaste={this.tryPaste}
-                    onKeyUp={this.handleCheckCaps}
-                    onKeyDown={this.handleCheckCaps}
+                    onKeyPress={this.handleCheckCaps}
```

Comment:
j/w how's this different from ` window.addEventListener('keydown', this.handleCheckCaps);` above in `componentWillMount`. and if it's the same, why are there two different listeners. shouldn't one be enough?

Reply:
Same question for `PasswordRepeatInput` below.

Reply:
One on window (or document.body) would be practically enough.  In theory somebody could use the component, listen for the same event and cancel it bubbling up. In this case we would not get any event on window (or document.body).

Reply:
I see. So the solution is to double-listen, on both `window` and the input field? :) There *has* to be a better way!
Why wouldn't listening only on the input field not be enough? I mean, why not remove `window.addEventListener('keydown', this.handleCheckCaps)` altogether.

Reply:
> Why wouldn't listening only on the input field not be enough?

because if the element has no focus it would not catch a capslock change, ideally it should always catch a capslock change EVEN when the app does not have focus, but that is not possible with JavaScript only.

Reply:
but we're not listening for the caps lock change anymore, are we? you even have a comment saying that. so, seems redundant.

Reply:
we try to catch every KeyboardEvent and try to if capslock is active. This would be no problem if `getModifierState` would work correctly, but in Qt it always return `false`

- https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent
- https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/getModifierState